net/instaweb/rewriter/csp.cc

```diff
@@ -49,43 +49,139 @@ inline bool IsAsciiAlpha(char ch) {
           ((ch >= 'A') && (ch <= 'Z')));
 }
 
+inline bool IsAsciiDigit(char ch) {
+  return ((ch >= '0') && (ch <= '9'));
+}
+
 }  // namespace
 
+bool CspSourceExpression::TryParseScheme(StringPiece* input) {
+  if (input->size() < 2) {
+    // Need at least a: or such.
+    return false;
+  }
+
+  if (!IsAsciiAlpha((*input)[0])) {
+    return false;
+  }
+
+  size_t pos = 1;
+  while (pos < input->size() &&
+         (IsAsciiAlphaNumeric((*input)[pos])
+          || (*input)[pos] == '+'
+          || (*input)[pos] == '-'
+          || (*input)[pos] == '.')) {
```

Comment:
it annoys me slightly that (*input)[pos] is repeated 4x :)

Maybe just factor out the per-char check to an inline helper?